packages/output/src/browser/output-widget.tsx

```diff
@@ -27,7 +28,8 @@ export const OUTPUT_WIDGET_KIND = 'outputView';
 @injectable()
 export class OutputWidget extends ReactWidget {
 
-    protected selectedChannel: OutputChannel | undefined;
+    public static selectedChannelEmitter: Emitter<void> = new Emitter<void>();
```

Comment:
Why is it static?

Reply:
The toolbar items are not scoped to the view lifecycle.  They exist outside it, though are visible only when the current view is an appropriate view.  So the emitter, to which the toolbar item listens, must also exist outside the view lifecycle.  If an OutputWidget is disposed, the toolbar item is still going to be listening to the emitter (though it won't receive any events if there is no OutputWidget to fire them).

The alternative is to create and dispose of the toolbar items along with the view.  This would involve registering them in the view init() instead of the contribution, and deregistering when the view is disposed.  I don't see this pattern used elsewhere but in other cases I have looked at, the toolbar action executes against a model and does not need a view widget.

Reply:
A better alternative would be to move the selectedChannel property from OutputWidget to OutputChannelManager.  The toolbar items can then operate on OuputChannelManager (the model) instead of the view widget.

Reply:
I generally don't like static members. It puts them outside of DI context and make impossible for extenders to change. It would be nice to turn into instance member. 

Encapsulating `selectedChannel` state and associated emitter from the widget to manager sounds like a good idea. Clients then can read/update the current channel and listen to its changes.

Reply:
I have now moved the channel selection from OutputWidget to OutputChannelManager.  The toolbar contributions now interact with OutputChannelManager directly and do not need an OutputWidget instance or the static which is now removed.  This does have the side-effect that closing and toggling back to the output view now restores the channel selection.

You will notice that the toolbar contributions are in a separate file.  I had originally used the pattern in problem-contribution with the view contribution and toolbar contribution being implemented in the same class.  However this caused problems in Inversify.

I believe all the issues are now addressed.